samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -19,36 +19,199 @@
 
 package org.apache.samza.runtime;
 
-import org.apache.samza.config.Config;
-import org.apache.samza.application.StreamApplication;
-import org.apache.samza.job.ApplicationStatus;
+  import java.util.HashMap;
+  import java.util.List;
+  import java.util.UUID;
+  import java.util.concurrent.CountDownLatch;
+  import java.util.concurrent.TimeUnit;
+  import java.util.concurrent.atomic.AtomicReference;
+  import org.apache.samza.SamzaException;
+  import org.apache.samza.application.StreamApplication;
+  import org.apache.samza.config.ApplicationConfig;
+  import org.apache.samza.config.Config;
+  import org.apache.samza.coordinator.CoordinationServiceFactory;
+  import org.apache.samza.coordinator.CoordinationUtils;
+  import org.apache.samza.coordinator.Latch;
+  import org.apache.samza.execution.ExecutionPlan;
+  import org.apache.samza.job.ApplicationStatus;
+  import org.apache.samza.processor.StreamProcessor;
+  import org.apache.samza.processor.StreamProcessorLifeCycleAware;
+  import org.apache.samza.system.StreamSpec;
+  import org.apache.samza.task.AsyncStreamTaskFactory;
+  import org.apache.samza.task.StreamTaskFactory;
+  import org.apache.samza.task.TaskFactoryUtil;
+  import org.apache.samza.util.ClassLoaderHelper;
+  import org.eclipse.jetty.util.ConcurrentHashSet;
+  import org.slf4j.Logger;
+  import org.slf4j.LoggerFactory;
 
 
 /**
  * This class implements the {@link ApplicationRunner} that runs the applications in standalone environment
  */
 public class LocalApplicationRunner extends AbstractApplicationRunner {
 
-  public LocalApplicationRunner(Config config) {
+  private static final Logger log = LoggerFactory.getLogger(LocalApplicationRunner.class);
+  private static final String LATCH_INIT = "init";
+  private static final long LATCH_TIMEOUT = 10; // 10 min timeout
+
+  private final String uid;
+  private final CoordinationUtils coordination;
+  private final CountDownLatch latch = new CountDownLatch(1);
+  private final ConcurrentHashSet<StreamProcessor> processors = new ConcurrentHashSet<>();
+  private final AtomicReference<Throwable> throwable = new AtomicReference<>();
+
+
+  private ApplicationStatus appStatus = ApplicationStatus.New;
+
+  final class LocalProcessorListener implements StreamProcessorLifeCycleAware {
+    StreamProcessor processor;
+
+    LocalProcessorListener(StreamProcessor processor) {
+      this.processor = processor;
+    }
+
+    @Override
+    public void onStart() {
```

Comment:
Shouldn't a processor be added to the set in onStart as opposed to when you are creating the StreamProcessors?